packages/turf-clean-coords/index.d.ts

```diff
@@ -0,0 +1,10 @@
+/// <reference types="geojson" />
+
+type Feature = GeoJSON.Feature<any> | GeoJSON.GeometryObject;
+
+/**
+ * http://turfjs.org/docs/#cleancoords
+ */
+declare function cleanCoords(feature: Feature, mutate?: boolean): Feature;
```

Comment:
@stebogit This one gets a bit tricky, one thing to keep in mind is trying to preserve the Input/Output type as much as possible.

At the moment, this definition is saying the input is any Feature or Geometry and it outputs either Geometry or a Feature.

Using the `extends` you validate the input, however the output will be the same as what the user enters, that way it preserves their Types.

```typescript
type GeometryObject = GeoJSON.GeometryObject
type Feature = GeoJSON.Feature<any>

/**
 * http://turfjs.org/docs/#cleancoords
 */
declare function cleanCoords<T extends GeometryObject|Feature>(feature: T, mutate?: boolean): T;
declare namespace cleanCoords {}
export = cleanCoords;
```